### PR TITLE
Warn when SGB compatibility is set without old licensee 0x33

### LINKED ISSUE #1150: Warn when `--sgb-compatible` is passed, but the old licensee code is not 0x33
Since RGBFIX sees all of the header anyway, we should be able to just check for this?

### PR DESCRIPTION
**Pull request by Rangi42**
Fixes #1150

There are three test cases related to this:
- `sgb` sets the SGB flag, does not specify the old licensee, but it's OK because the old licensee was already 0x33 (it wasn't before but this PR makes it so)
- `sgb-licensee` sets the SGB flag, and specifies the old licensee as 0x69, so it gets a warning
- `sgb-old-licensee` sets the SGB flag, does not specify the old licensee, and gets a warning because the old licensee was 0xc5 (new test case for this PR)

### CODE REVIEW

File: src/fix/main.c [hunk 1]
```diff
@@ -967,6 +967,10 @@ static void processFile(int input, int output, char const *name, off_t fileSize)
 
 	if (oldLicensee != UNSPECIFIED)
 		overwriteByte(rom0, 0x14B, oldLicensee, "old licensee code");
+	else if (sgb && rom0[0x14B] != 0x33)
+		fprintf(stderr,
+			"warning: SGB compatibility enabled, but old licensee was %#x, not 0x33\n",
```
Comment (ISSOtm) [inline src/fix/main.c:972]:
```suggestion
			"warning: SGB compatibility enabled, but old licensee was 0x%02x, not 0x33\n",
```

**Review by ISSOtm — DISMISSED** [review]
Oh cool, that was much simpler than I expected!

File: src/fix/main.c (hunk 1 above)
Comment (Rangi42) [reply src/fix/main.c:972]: We have 11 instances of `%#` and 6 instances of `0x%`. Would you like to standardize on one or the other?

Comment (ISSOtm) [reply src/fix/main.c:972]: `%#` is amenable to neither a custom prefix (`$%x`) nor to a clear min width (`0x%02x`), so it shouldn't be used, IMO.

**Review by ISSOtm — APPROVED** [review]

**Pull request merged**